### PR TITLE
feat(console): sign-in-methods form setup

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add form fields for "sign in methods" tab in page: sign in experience.

Please notice that there are still many issues in this page, this PR focused on "setup" of fields and data models.

## Linear Issue Reference

LOG-1730

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/utilities.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  return SignInMethodState.disabled;
+};
+
+export const signInExperienceParser = {
+  toLocalForm: (signInExperience: SignInExperience): SignInExperienceForm => {
+    const methodKeys: Array<keyof SignInMethods> = ['username', 'email', 'sms', 'social'];
+    const primaryMethod = methodKeys.find(
+      (key) => signInExperience.signInMethods[key] === 'primary'
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/utilities.ts:26]: better `signInExperience.signInMethods[key] === SignInMethodState.primary`?

File: packages/console/src/pages/SignInExperience/utilities.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    );
+    const secondaryMethods = methodKeys.filter(
+      (key) => signInExperience.signInMethods[key] === 'secondary'
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/utilities.ts:29]: ditto
better `signInExperience.signInMethods[key] === SignInMethodState.second`?

File: packages/console/src/pages/SignInExperience/utilities.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  if (signInMethods[method]) {
+    return SignInMethodState.secondary;
+  }
+
+  return SignInMethodState.disabled;
+};
+
+export const signInExperienceParser = {
+  toLocalForm: (signInExperience: SignInExperience): SignInExperienceForm => {
+    const methodKeys: Array<keyof SignInMethods> = ['username', 'email', 'sms', 'social'];
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/utilities.ts:24]: nit: better extract `SignInMethodKey`?
image

and then you can use
`const methodKeys = Object.values(SignInMethodKey);`

Comment (IceHe) [reply packages/console/src/pages/SignInExperience/utilities.ts:24]: @wangsijie PTAL #556

Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/utilities.ts:24]: updated

File: packages/console/src/pages/SignInExperience/utilities.ts (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/utilities.ts:26]: updated

File: packages/console/src/pages/SignInExperience/utilities.ts (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/utilities.ts:29]: updated

**Review by IceHe — APPROVED** [review]
LGTM

**Pull request merged**